src/sage/schemes/elliptic_curves/mod_poly.py

```diff
@@ -0,0 +1,166 @@
+        1.84 µs ± 1.84 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
+        sage: %timeit classical_modular_polynomial(101, GF(65537).random_element())  # not tested
+        59.8 ms ± 29.4 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
+
+    TESTS::
+
+        sage: q = random_prime(50)^randrange(1,4)
+        sage: j = GF(q).random_element()
+        sage: l = random_prime(50)
+        sage: Y = polygen(parent(j), 'Y')
+        sage: classical_modular_polynomial(l,j) == classical_modular_polynomial(l)(j,Y)
+        True
+    """
+    l = ZZ(l)
+
+    if j is None:
+        # We are supposed to return the generic modular polynomial. First
+        # check if it is already in the cache, then check the database,
+        # finally compute it using PARI.
+        try:
+            return _cache[l]
+        except KeyError:
+            pass
+
+        try:
+            Phi = ZZ['X,Y'](_db[l])
+        except ValueError:
+            try:
+                pari_Phi = pari.polmodular(l)
+            except PariError:
+                raise NotImplementedError('modular polynomial is not in database and computing it on the fly is not yet implemented')
+            d = {(i, j): c for i,f in enumerate(pari_Phi) for j, c in enumerate(f)}
+            Phi = ZZ['X,Y'](d)
+
+        if l <= _cache_bound:
+            _cache[l] = Phi
+
+        return Phi
+
+    R = parent(j)['Y']
+    Y = R.gen()
+
+    # If the generic polynomial is in the cache or the database, evaluating
+    # it directly should always be faster than recomputing it from scratch.
+    if l  in _cache:
+        return _cache[l](j, Y)
+    try:
+        Phi = _db[l]
+    except ValueError:
+        pass
+    else:
+        if l <= _cache_bound:
+            _cache[l] = ZZ['X,Y'](Phi)
+        return Phi(j, Y)
+
+    # Now try to get the instantiated modular polynomial directly from PARI.
+    # This should be slightly more efficient (in particular regarding memory
+    # usage) than computing and evaluating the generic modular polynomial.
+    try:
+        pari_Phi = pari.polmodular(l, 0, j)
+    except PariError:
+        pass
+    else:
+        return R(pari_Phi)
+
+    # Nothing worked. Fall back to computing the generic modular polynomial
+    # and simply evaluating it.
+    return classical_modular_polynomial(l)(j, Y)
+
+def _set_cache_bound(bnd):
+    r"""
+    Internal helper function to allow setting the caching cutoff for
+    :func:`classical_modular_polynomial`.
+
+    Exposed as ``classical_modular_polynomial.set_cache_bound()``.
+
+    EXAMPLES::
+
+        sage: import sage.schemes.elliptic_curves.mod_poly as m
+        sage: m._cache_bound
+        100
+        sage: m._set_cache_bound(123)
+        sage: m._cache_bound
+        123
+        sage: classical_modular_polynomial.set_cache_bound is m._set_cache_bound
+        True
+    """
+    global _cache_bound
+    _cache_bound = bnd
+classical_modular_polynomial.set_cache_bound = _set_cache_bound
```

Comment:
```suggestion
    _cache_bound = bnd

classical_modular_polynomial.set_cache_bound = _set_cache_bound
```
Linter thing. Once done, you can set a positive review. Sorry for the last change needed.